database_client/result_formatter.py

```diff
@@ -101,9 +106,27 @@ def format_result_with_subquery_parameters(
     @staticmethod
     def format_with_metadata(
         data: list[dict],
-    ) -> dict:
+        relation_name: str,
+        subquery_parameters: Optional[list[SubqueryParameters]] = [],
+    ) -> dict[str, Any]:
+        metadata_dict = {}
+        relation_reference = SQL_ALCHEMY_TABLE_REFERENCE[relation_name]
+
+        # Iterate through all properties of the Table
+        for name, descriptor in inspect(
+            relation_reference
+        ).all_orm_descriptors.items():
+            if type(descriptor) == hybrid_method and name in METADATA_METHOD_NAMES:
```

Comment:
Nit: In a situation like this, don't forget about [guard clauses](https://en.wikipedia.org/wiki/Guard_(computer_science)). In this case, it's a bit more challenging because there's two logical conditionals, but we can apply [DeMorgan's law](https://en.wikipedia.org/wiki/De_Morgan%27s_laws) and invert both:
```python
if type(descriptor) != hybrid_method or name not in METADATA_METHOD_NAMES:
   continue
## Perform the rest of the logic
```